src/SpinWaveTheory/SWTCalculations.jl

```diff
@@ -53,33 +64,68 @@ function swt_hamiltonian_SUN!(swt::SpinWaveTheory, q_reshaped::Vec3, Hmat::Matri
             ### Bilinear exchange
             
             J = Mat3(coupling.bilin*I)
+            
             sub_i, sub_j = bond.i, bond.j
+            sub_i_M1, sub_j_M1 = sub_i - 1, sub_j - 1
+            
             phase = exp(2π*im * dot(q_reshaped, bond.n)) # Phase associated with periodic wrapping
 
-            tTi_μ = s̃_mat[:, :, :, sub_i]
-            tTj_ν = s̃_mat[:, :, :, sub_j]
-            sub_i_M1, sub_j_M1 = sub_i - 1, sub_j - 1
+            # Fill dipole part of tTi_μ
+            tTi_μ[:, :, 1:3] .= view(s̃_mat,:, :, 1:3, sub_i)
+            tTj_ν[:, :, 1:3] .= view(s̃_mat,:, :, 1:3, sub_j)
+            
+            # Fill quadrupole part of tTi_μ
+            tTi_μ[:, :, 4:8] .= view(Q̃_mat,:, :, 1:5, sub_i)
+            tTj_ν[:, :, 4:8] .= view(Q̃_mat,:, :, 1:5, sub_j)
```

Comment:
@Hao-Phys this the main thing that it would be good to check on; are these comments accurate? and what exactly are the `tTi_mu` and `t_mu_ii` variables meaning here?

Reply:
These comments are correct. `tTi_mu` means the SU(N) generators in the local frame at site `i`, and `tTj_mu` means the same but at site `j`. Here sites `i` and `j` are connected by a bond.